src/components/select/select.ts

```diff
@@ -451,9 +490,15 @@ export class Select extends BaseInput<any> implements OnDestroy {
     if (this._options) {
       this._options.forEach(option => {
         // check this option if the option's value is in the values array
-        option.selected = this.getValues().some(selectValue => {
-          return isCheckedProperty(selectValue, option.value);
-        });
+        if (this._compareWith) {
+          option.selected = this.getValues().some(selectValue => {
+            return this._compareWith(selectValue, option.value);
+          });
+        } else {
+          option.selected = this.getValues().some(selectValue => {
+            return isCheckedProperty(selectValue, option.value);
```

Comment:
what is  isCheckedProperty is the default, compare function? this way we don't have to duplicate code. does it make sense?